source/NVDAMagnifier/__init__.py

```diff
@@ -0,0 +1,842 @@
+import ctypes
+from ctypes import wintypes
+from enum import Enum
+from typing import Callable
+
+from .windowsHandler import DockedFrame, LensFrame
+
+from logHandler import log
+import ui
+import wx
+import api
+
+# Utils
+
+
+class MouseHandler:
+	def __init__(self):
+		self._mousePosition: tuple[int, int] = (0, 0)
+
+	@property
+	def mousePosition(self):
+		return self.getMousePosition()
+
+	@mousePosition.setter
+	def mousePosition(self, pos: tuple[int, int]):
+		self._mousePosition = pos
+
+	def getMousePosition(self) -> tuple[int, int]:
+		"""
+		Get the current mouse position as (x, y).
+		"""
+		pt = wintypes.POINT()
+		ctypes.windll.user32.GetCursorPos(ctypes.byref(pt))
+		return (pt.x, pt.y)
+
+	def isLeftClickPressed(self) -> bool:
+		"""
+		Check if the left mouse button is currently pressed.
+		"""
+		# VK_LBUTTON = 0x01 (Virtual key code for left mouse button)
+		# GetKeyState returns negative value if key is pressed
+		return ctypes.windll.user32.GetKeyState(0x01) < 0
+
+
+class ColorFilter(Enum):
+	NORMAL = "normal"
+	GREYSCALE = "greyscale"
+	INVERTED = "inverted"
+
+
+class FullScreenMode(Enum):
+	CENTER = "center"
+	BORDER = "border"
+	RELATIVE = "relative"
+
+
+class MagnifierType(Enum):
+	FULLSCREEN = "fullscreen"
+	DOCKED = "docked"
+	LENS = "lens"
+
+
+class ColorFilterMatrix(Enum):
+	NORMAL = (ctypes.c_float * 25)(
+		1.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		1.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		1.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		1.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		1.0,
+	)
+
+	GREYSCALE = (ctypes.c_float * 25)(
+		0.33,
+		0.33,
+		0.33,
+		0.0,
+		0.0,
+		0.59,
+		0.59,
+		0.59,
+		0.0,
+		0.0,
+		0.11,
+		0.11,
+		0.11,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		1.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		1.0,
+	)
+
+	INVERTED = (ctypes.c_float * 25)(
+		-1.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		-1.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		-1.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		1.0,
+		0.0,
+		1.0,
+		1.0,
+		1.0,
+		0.0,
+		1.0,
+	)
+
+
+# Base Code
```

Comment:
these kinds of comments can largely be removed. code grouping should be largely done through separate files